hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/WriteJdbcP.java

```diff
@@ -0,0 +1,201 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.impl.connector;
+
+import com.hazelcast.jet.core.Inbox;
+import com.hazelcast.jet.core.Outbox;
+import com.hazelcast.jet.core.Processor;
+import com.hazelcast.jet.core.ProcessorMetaSupplier;
+import com.hazelcast.jet.core.processor.SinkProcessors;
+import com.hazelcast.jet.function.DistributedBiConsumer;
+import com.hazelcast.jet.function.DistributedFunction;
+import com.hazelcast.jet.function.DistributedSupplier;
+import com.hazelcast.jet.impl.util.ExceptionUtil;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.util.concurrent.BackoffIdleStrategy;
+import com.hazelcast.util.concurrent.IdleStrategy;
+
+import javax.annotation.Nonnull;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.SQLNonTransientException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.hazelcast.jet.impl.util.Util.uncheckCall;
+import static com.hazelcast.jet.impl.util.Util.uncheckRun;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * Private API, use {@link SinkProcessors#writeJdbcP}.
+ */
+public final class WriteJdbcP<T> implements Processor {
+
+    private static final IdleStrategy IDLER =
+            new BackoffIdleStrategy(0, 0, MILLISECONDS.toNanos(1), SECONDS.toNanos(10));
+
+    private final DistributedSupplier<Connection> connectionSupplier;
+    private final DistributedFunction<Connection, PreparedStatement> statementFn;
+    private final DistributedBiConsumer<PreparedStatement, T> updateFn;
+    private final DistributedBiConsumer<Connection, PreparedStatement> flushFn;
+
+    private ILogger logger;
+    private Connection connection;
+    private PreparedStatement statement;
+    private List<T> itemList = new ArrayList<>();
+    private int idleCount;
+
+    public WriteJdbcP(
+            @Nonnull DistributedSupplier<Connection> connectionSupplier,
+            @Nonnull DistributedFunction<Connection, PreparedStatement> statementFn,
+            @Nonnull DistributedBiConsumer<PreparedStatement, T> updateFn,
+            @Nonnull DistributedBiConsumer<Connection, PreparedStatement> flushFn
+    ) {
+        this.connectionSupplier = connectionSupplier;
+        this.statementFn = statementFn;
+        this.updateFn = updateFn;
+        this.flushFn = flushFn;
+    }
+
+    @Override
+    public void init(@Nonnull Outbox outbox, @Nonnull Context context) {
+        logger = context.logger();
+        connection = connectionSupplier.get();
```

Comment:
We should not fail on connection error here too. We can assign `idleCount=1` and call `reconnectIfNecessary`